### PR TITLE
Make creation of secret configurable

### PR DESCRIPTION
**Pull request by andor44**
This way users can supply their own secrets so it doesn't have to be stored in a values.yaml

**What this PR does / why we need it**:
It makes creation of the secret object as part of the chart installation optional.

In our deployment workflows secrets aren't created by helm, they're created separately. I imagine that this is quite a common pattern, as it is not very idiomatic to put secrets into your values file (which is likely checked into source control).

### CODE REVIEW

**Review by nitisht — COMMENTED** [review]
@andor44 LGTM, can you add examples on how to use `existingSecretName`

**Comment by andor44** [comment]
Added usage example in README

**Comment by andor44** [comment]
Can I get an ok-to-test please?

**Review by mlaccetti — APPROVED** [review]
/lgtm

File: stable/minio/templates/deployment.yaml [hunk 1]
```diff
@@ -78,12 +78,12 @@ spec:
             - name: MINIO_ACCESS_KEY
               valueFrom:
                 secretKeyRef:
-                  name: {{ template "minio.fullname" . }}
+                  name: {{ if .Values.createSecret }}{{ template "minio.fullname" . }}{{ else }}{{ .Values.existingSecretName }}{{ end }}
```
Comment (unguiculus) [inline stable/minio/templates/deployment.yaml]: Please drop `createSecret`, rename `existingSecretName` to `existingSecret`, and use it if set. You can find many examples that do it that way. Please follow this style. Also, please create a helper template for the secret name.

Here's an example:
https://github.com/helm/charts/blob/cf7a5644c63d7481528206fe59939ef4ae3a119e/stable/openldap/templates/_helpers.tpl#L35-L40
https://github.com/helm/charts/blob/cf7a5644c63d7481528206fe59939ef4ae3a119e/stable/openldap/templates/deployment.yaml#L43

**Review by unguiculus — DISMISSED** [review]

File: stable/minio/templates/deployment.yaml (hunk 1 above)
Comment (andor44) [reply stable/minio/templates/deployment.yaml]: I vaguely recalled having a reason for using 2 variables but I don't know what it was. Either, way updated, only using a single variable now!

**Review by scottrigby — APPROVED** [review]
Dismissing https://github.com/helm/charts/pull/6789#pullrequestreview-144513410 - it was resolved in 2f55d53c27d435bdf1bfae7b90fff2d6f0177367 👍

**Pull request merged**